app/models/tagset_models/tag_set.rb

```diff
@@ -250,17 +271,21 @@ def remove_from_autocomplete
 
   def add_tags_to_autocomplete(tags_to_add)
     tags_to_add.each do |tag|
+      next if TagSet.invalid_tag?(tag.name) # skips invalid tags in autocomplete
       value = tag.autocomplete_value
       REDIS_AUTOCOMPLETE.zadd("autocomplete_tagset_all_#{self.id}", 0, value)
       REDIS_AUTOCOMPLETE.zadd("autocomplete_tagset_#{tag.type.downcase}_#{self.id}", 0, value)
     end
   end
 
-  def remove_tags_from_autocomplete(tags_to_remove)
-    tags_to_remove.each do |tag|
-      value = tag.autocomplete_value
-      REDIS_AUTOCOMPLETE.zrem("autocomplete_tagset_all_#{self.id}", value)
-      REDIS_AUTOCOMPLETE.zrem("autocomplete_tagset_#{tag.type.downcase}_#{self.id}", value)
+  # cleanup of invalid tags for autocomplete
+  def remove_invalid_tags_from_autocomplete
```

Comment:
This method can't just be repurposed like this, it's used for removing tags from the autocomplete when removing them from the tag set.

Instead, to do the cleanup suggested in the Jira issue, you'd have to write a rake task that goes through all tag sets and removes the invalid tags from the tag set's autocomplete. This is optional according to the Jira issue, so feel free to leave it out.

If you do want to create the cleanup task, there is even more to be done. The current code adds the invalid tags to the `set_taggings` of the tag set, with a nil tag on the set_tagging. So the cleanup code should remove those set_taggings as well.